### PR TITLE
enabled avoid_dynamic_calls check for post related functionalities

### LINKED ISSUE #1761: Enabling `avoid_dynamic_calls` for post related functionalities
**Describe the bug**
We need to re-enable the `avoid_dynamic_calls` check and fix the resulting bugs. However since the bugs are highly sensitive, we are creating multiple issues for different functionalities so that it becomes easier to spot any unforeseen problems.

**To Reproduce**
Run the current suite of tests, and they all pass. This is because the avoid_dynamic_calls check has been disabled

**Expected behavior**
`avoid_dynamic_calls` must be re-enabled, and all the resulting errors must be fixed.

**Actual behavior**
`avoid_dynamic_calls` is currently disabled.

**Additional details**
This is a sub-part of issue #1721 for all the post-related bugs.

**Potential internship candidates**
Please read this if you are planning to apply for a Palisadoes Foundation internship https://github.com/PalisadoesFoundation/talawa/issues/359

### PR DESCRIPTION
**Pull request by Amogh35**
**What kind of change does this PR introduce?**

bugfix

**Issue Number:**

Fixes #1761  

**Did you add tests for your changes?**

NA

**Snapshots/Videos:**

NA

**Summary**

Type casted the resulting bugs In post_model.dart and post_service.dart. Had to make necessary changes in chat_service.dart and user_config.dart to accommodate the previously made changes

**Does this PR introduce a breaking change?**

No

**Other information**

This is an alternative PR for #1762 as the previous one became messy with too many conflicts

**Have you read the [contributing guide](https://github.com/PalisadoesFoundation/talawa/blob/master/CONTRIBUTING.md)?**

Yes

### CODE REVIEW

**Comment by palisadoes** [comment]
Please fix the failing tests. Our test code coverage system will fail if any of these conditions occur:

1) The overall code coverage drops below the target threshold of the repository
2) Any file in the pull request has code coverage levels below the repository threshold
3) The code does not comply with our linting check or other related formatting requirements 

We do this to ensure the overall reliability of the code base is maintained. This is a prerequisite for getting your PR approved. Assigned reviewers regularly review the PR queue and tend to focus on PRs that are passing.

File: lib/services/chat_service.dart [hunk 1]
```diff
@@ -1,5 +1,4 @@
-// ignore_for_file: talawa_api_doc, avoid_dynamic_calls
-// ignore_for_file: talawa_good_doc_comments
+// ignore_for_file: avoid_dynamic_calls
```
Comment (Ayush0Chaudhary) [inline lib/services/chat_service.dart:1]: @Amogh35 I think you forgot to enable dynamic calls to this file

File: lib/services/user_config.dart [hunk 2]
```diff
@@ -1,6 +1,4 @@
-// ignore_for_file: talawa_api_doc, avoid_dynamic_calls
-// ignore_for_file: talawa_good_doc_comments
-
+// ignore_for_file: avoid_dynamic_calls
```
Comment (Ayush0Chaudhary) [inline lib/services/user_config.dart:1]: this too

**Comment by Ayush0Chaudhary** [comment]
Apologies for meddling in this PR but I am working on chatservice functionality for this issue https://github.com/PalisadoesFoundation/talawa/issues/1748

**Comment by Amogh35** [comment]
> Please fix the failing tests. Our test code coverage system will fail if any of these conditions occur:
> 
> 1. The overall code coverage drops below the target threshold of the repository
> 2. Any file in the pull request has code coverage levels below the repository threshold
> 3. The code does not comply with our linting check or other related formatting requirements
> 
> We do this to ensure the overall reliability of the code base is maintained. This is a prerequisite for getting your PR approved. Assigned reviewers regularly review the PR queue and tend to focus on PRs that are passing.

Yes, I will start fixing the tests right away.

**Comment by Amogh35** [comment]
> Apologies for meddling in this PR but I am working on chatservice functionality for this issue #1748

No worries, Ayush; the resulting bugs in this issue are very sensitive and are frequently causing the functionality to collapse.
This is why I was planning to fix the main functionalities in different PRs so that we can quickly troubleshoot any problem we face in the future.
If it obstructs your PR, I can start working on the chat functionality immediately after I finish this PR.

**Comment by Amogh35** [comment]
@palisadoes Sir, I will be fixing well over 24 files for this issue #1721  and also adding documentation for each one of them. I believe adding tests for all these changed files is vastly outside the scope of this issue, as creating tests for all the changes in over 24 files is quite a huge task.
Moreover, the codecov report shows that this will not change the coverage. So could you please consider the circumstances and suggest an appropriate course of action?

**Comment by palisadoes** [comment]
1. I understand your concern about writing tests, it is an arduous activity especially when the original files were not created by you. 
2. However, if we don't try to enforce a testing regime we will be back to where our code consistently malfunctioned.
3. In the other repositories the contributors focused on getting the tests working in which we discovered many inconsistencies that have made the code quality better. This repository has lagged.
4. We need to get the coverage up to a standard for having an MVP this year.

I know that tests can be written to cheat the system, but overall our contributors have tried to make the code better. Please try your best to improve our code quality. It is the sole reason for creating the issue for this PR after doing the flutter code upgrade.

**Comment by palisadoes** [comment]
How's the progress on this?

**Comment by Amogh35** [comment]
I am still working on fixing the tests. Unfortunately, my End Semester exams have just started today, and I will not be able to work on this issue as actively.
I realize that I am blocking others from gaining experience and fixing this important bug, hence I am willing to step aside if someone else is willing to work on this issue more actively.
Instead, I can come back after a week, once my exams are over, and resume working on whatever issue needs to be fixed.
Hope you can understand my situation.

**Comment by palisadoes** [comment]
Should this be closed?

**Comment by palisadoes** [comment]
Closing due to inactivity

**Pull request closed without merging**